be1-go/hub/standard_hub/message_handling.go

```diff
@@ -269,6 +269,16 @@ func (h *Hub) handlePublish(socket socket.Socket, byteMessage []byte) (int, erro
 		return -1, xerrors.Errorf("failed to unmarshal publish message: %v", err)
 	}
 
+	signature := publish.Params.Message.Signature
+	messageId := publish.Params.Message.MessageID
+	data := publish.Params.Message.Data
+
+	expectedMessageId := messagedata.Hash(data, signature)
```

Comment:
From a security perspective, there should be a delimiter between data and signature. https://crypto.stackexchange.com/questions/55162/best-way-to-hash-two-values-into-one

Reply:
Although this change would likely require the cooperation of all subsystems but generally its essential to add a delimiter when concatenating strings that is to be hashed.

Reply:
I see I see, for the moment the delimiter is not included in any subsystem if I am not mistaken, so this should be a global change in the system and I guess in the specifications as well

Reply:
Actually, we don't need a delimiter because we prefix each data element with its length.